base/uk.ac.stfc.isis.ibex.ui.widgets/src/uk/ac/stfc/isis/ibex/ui/widgets/SaveScriptGeneratorFileMessageDialog.java

```diff
@@ -0,0 +1,324 @@
+package uk.ac.stfc.isis.ibex.ui.widgets;
+
+import static org.eclipse.swt.events.SelectionListener.widgetSelectedAdapter;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FilenameFilter;
+import java.io.IOException;
+
+import org.apache.logging.log4j.Logger;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+import uk.ac.stfc.isis.ibex.logger.IsisLog;
+
+
+/**
+ * A message dialog box that allows you to copy the contents of the dialog and open the file the dialog is tied to.
+ * 
+ * @author https://stackoverflow.com/a/30630475 modified by James King
+ *
+ */
+public class SaveScriptGeneratorFileMessageDialog extends MessageDialog {
```

Comment:
There is a lot of logic in this class which shouldn't exist in the UI layer/view. Generating the filename, checking files on disc and saving a file are more appropriately placed near the script gen singleton, in the 'model'. Is it possible to move these methods to there? Not having this logic in the UI layer would also make it easier to write/run tests.

Reply:
I agree, moved out to ScriptGeneratorFileHandler